modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/session/v1/UserSessionTest.java

```diff
@@ -0,0 +1,252 @@
+
+        isURL = backendURL.substring(0, backendURL.indexOf("services/"));
+
+        serviceProviderTravelocity = createServiceProvider(ISSUER_TRAVELOCITY_COM, SERVICE_PROVIDER_NAME_TRAVELOCITY);
+        serviceProviderAvis = createServiceProvider(ISSUER_AVIS_COM, SERVICE_PROVIDER_NAME_AVIS);
+
+        createUsersForTesting();
+        createSessionsForTesting();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testConclude() throws Exception {
+
+        super.conclude();
+        appMgtclient.deleteApplication(serviceProviderTravelocity.getApplicationName());
+        appMgtclient.deleteApplication(serviceProviderAvis.getApplicationName());
+        userMgtClient.deleteUser(session_test_user1);
+        userMgtClient.deleteUser(session_test_user2);
+    }
+
+    private SAMLSSOServiceProviderDTO createSAMLServiceProviderDTO(String issuer) {
+
+        SAMLSSOServiceProviderDTO samlssoServiceProviderDTO = new SAMLSSOServiceProviderDTO();
+        samlssoServiceProviderDTO.setIssuer(issuer);
+        samlssoServiceProviderDTO.setAssertionConsumerUrls(new String[]
+                {String.format(SAMPLE_APP_URL, issuer) + "/home.jsp"});
+        samlssoServiceProviderDTO.setDefaultAssertionConsumerUrl(String.format(SAMPLE_APP_URL, issuer) +
+                "/home.jsp");
+        samlssoServiceProviderDTO.setAttributeConsumingServiceIndex("1239245949");
+        samlssoServiceProviderDTO.setNameIDFormat("urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress");
+        samlssoServiceProviderDTO.setDoSignAssertions(true);
+        samlssoServiceProviderDTO.setDoSignResponse(true);
+        samlssoServiceProviderDTO.setDoSingleLogout(true);
+        samlssoServiceProviderDTO.setLoginPageURL("/carbon/admin/login.jsp");
+        return samlssoServiceProviderDTO;
+    }
+
+    private void createUsersForTesting() throws RemoteException, UserAdminUserAdminException {
+
+        userMgtClient.addUser(session_test_user1, TEST_USER_PASSWORD, new String[]{ADMIN_ROLE}, DEFAULT_PROFILE);
+        userMgtClient.addUser(session_test_user2, TEST_USER_PASSWORD, null, DEFAULT_PROFILE);
+    }
+
+    private void createSessionsForTesting() throws Exception {
+
+        client = HttpClientBuilder.create().build();
+        loginUserToApplication(client, ISSUER_TRAVELOCITY_COM, session_test_user1, TEST_USER_PASSWORD);
+        loginUserToApplication(client, ISSUER_AVIS_COM, session_test_user1, TEST_USER_PASSWORD);
+        client.close();
+
+        client = HttpClientBuilder.create().build();
+        loginUserToApplication(client, ISSUER_TRAVELOCITY_COM, session_test_user1, TEST_USER_PASSWORD);
+        client.close();
+    }
+
+    private void loginUserToApplication(CloseableHttpClient client, String application, String username,
+                                        String password) throws Exception {
+
+        HttpPost request = new HttpPost(String.format(SAMPLE_APP_URL, application) + "/samlsso" +
```

Comment:
Move "/samlsso" and similar strings to constants

Reply:
fixed